internal/controller/storageclient_controller.go

```diff
@@ -386,82 +392,31 @@ func (r *StorageClientReconciler) onboardConsumer(externalClusterClient *provide
-		r.Log.Error(err, "empty response")
+		r.log.Error(err, "empty response")
 		return err
 	}
 
 	r.storageClient.Status.ConsumerID = response.StorageConsumerUUID
 	r.storageClient.Status.Phase = v1alpha1.StorageClientConnected
 
-	r.Log.Info("onboarding completed")
+	r.log.Info("onboarding completed")
 	return nil
 }
 
 // offboardConsumer makes an API call to the external storage provider cluster for offboarding
-func (r *StorageClientReconciler) offboardConsumer(externalClusterClient *providerClient.OCSProviderClient) (reconcile.Result, error) {
-
-	_, err := externalClusterClient.OffboardConsumer(r.ctx, r.storageClient.Status.ConsumerID)
-	if err != nil {
-		if st, ok := status.FromError(err); ok {
-			r.logGrpcErrorAndReportEvent(OffboardConsumer, err, st.Code())
-		}
-		return reconcile.Result{}, fmt.Errorf("failed to offboard consumer: %v", err)
-	}
-
-	return reconcile.Result{}, nil
-}
-
-func (r *StorageClientReconciler) logGrpcErrorAndReportEvent(grpcCallName string, err error, errCode codes.Code) {
-
-	var msg, eventReason, eventType string
-
-	if grpcCallName == OnboardConsumer {
-		if errCode == codes.InvalidArgument {
-			msg = "Token is invalid. Verify the token again or contact the provider admin"
-			eventReason = "TokenInvalid"
-			eventType = corev1.EventTypeWarning
-		} else if errCode == codes.AlreadyExists {
-			msg = "Token is already used. Contact provider admin for a new token"
-			eventReason = "TokenAlreadyUsed"
-			eventType = corev1.EventTypeWarning
-		}
-	} else if grpcCallName == OffboardConsumer {
-		if errCode == codes.InvalidArgument {
-			msg = "StorageConsumer UID is not valid. Contact the provider admin"
-			eventReason = "UIDInvalid"
-			eventType = corev1.EventTypeWarning
-		}
-	} else if grpcCallName == GetStorageConfig {
-		if errCode == codes.InvalidArgument {
-			msg = "StorageConsumer UID is not valid. Contact the provider admin"
-			eventReason = "UIDInvalid"
-			eventType = corev1.EventTypeWarning
-		} else if errCode == codes.NotFound {
-			msg = "StorageConsumer UID not found. Contact the provider admin"
-			eventReason = "UIDNotFound"
-			eventType = corev1.EventTypeWarning
-		} else if errCode == codes.Unavailable {
-			msg = "StorageConsumer is not ready yet. Will requeue after 5 second"
-			eventReason = "NotReady"
-			eventType = corev1.EventTypeNormal
-		}
-	}
-
-	if msg != "" {
-		r.Log.Error(err, "StorageProvider:"+grpcCallName+":"+msg)
-		r.recorder.ReportIfNotPresent(r.storageClient, eventType, eventReason, msg)
+func (r *storageClientReconcile) offboardConsumer(externalClusterClient *providerClient.OCSProviderClient) error {
+	if _, err := externalClusterClient.OffboardConsumer(r.ctx, r.storageClient.Status.ConsumerID); err != nil {
+		return fmt.Errorf("failed to offboard consumer: %v", err)
 	}
+	return nil
 }
 
-func (r *StorageClientReconciler) reconcileClientStatusReporterJob() (reconcile.Result, error) {
+func (r *storageClientReconcile) reconcileClientStatusReporterJob() error {
```

Comment:
We should return a result object as well.

Reply:
done.